facade/src/main/java/org/jboss/pnc/facade/providers/ArtifactProviderImpl.java

```diff
@@ -91,14 +119,26 @@ public ArtifactProviderImpl(
     @Override
     @RolesAllowed(SYSTEM_USER)
     public org.jboss.pnc.dto.Artifact store(org.jboss.pnc.dto.Artifact restEntity) throws DTOValidationException {
-        return super.store(restEntity);
+        org.jboss.pnc.model.User currentUser = userService.currentUser();
+        User user = userMapper.toDTO(currentUser);
+        Instant now = Instant.now();
+        return super.store(
+                restEntity.toBuilder()
+                        .creationUser(user)
+                        .modificationUser(user)
+                        .creationTime(now)
+                        .modificationTime(now)
+                        .build());
     }
 
     @Override
     @RolesAllowed(SYSTEM_USER)
     public org.jboss.pnc.dto.Artifact update(String id, org.jboss.pnc.dto.Artifact restEntity)
             throws DTOValidationException {
-        return super.update(id, restEntity);
+        org.jboss.pnc.model.User currentUser = userService.currentUser();
+        User user = userMapper.toDTO(currentUser);
+        Instant now = Instant.now();
+        return super.update(id, restEntity.toBuilder().modificationUser(user).modificationTime(now).build());
```

Comment:
Do we want to create new revision when the Quality level doesn't change? That means there will be 2 revisions only different with `modificationUser` and `modificationTime` but nothing else changed.

Reply:
Right, but in the datastore/src/main/java/org/jboss/pnc/datastore/repositories/ArtifactRepositoryImpl.java in the save method, if there are no changes to the audited values the modificationUser and time are reset, and no revision is created. I guess I could also revert the logic (only set modification time and users only if audited fields changed, in the datastore) if you prefer